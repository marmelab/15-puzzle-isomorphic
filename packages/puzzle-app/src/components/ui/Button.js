import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';

import { Link } from '../../routes';

const Button = ({ className, color, icon, label, onClick, route, style }) => {
    const buttonClass = ClassNames('btn', color ? color : '', className);

    if (route) {
        return (
            <Link href="" route={route}>
                <a className={buttonClass} title={label} style={style}>
                    <i className="material-icons left">{icon}</i>
                    {label}
                </a>
            </Link>
        );
    }
    return (
        <a
            onClick={onClick}
            className={buttonClass}
            title={label}
            style={style}
        >
            <i className="material-icons left">{icon}</i>
            {label}
        </a>
    );
};

Button.propTypes = {
    className: PropTypes.string,
    color: PropTypes.string,
    icon: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    route: PropTypes.string,
    style: PropTypes.object,
};

export default Button;
