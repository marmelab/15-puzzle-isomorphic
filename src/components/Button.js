import React from 'react';
import { Link } from '../routes';

import PropTypes from 'prop-types';
import ClassNames from 'classnames';

const Button = ({ className, color, icon, label, onClick, route }) => {
    const buttonClass = ClassNames(
        {
            btn: true,
            [color]: color,
            'waves-effect': true,
            'waves-light': true,
        },
        className,
    );

    if (route) {
        return (
            <Link href="" route={route}>
                <a className={buttonClass} title={label}>
                    <i className="material-icons left">{icon}</i>
                    {label}
                </a>
            </Link>
        );
    }
    return (
        <a onClick={onClick} className={buttonClass} title={label}>
            <i className="material-icons left">{icon}</i>
            {label}
        </a>
    );
};

Button.propTypes = {
    className: PropTypes.string,
    color: PropTypes.string,
    icon: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    route: PropTypes.string,
};

export default Button;
