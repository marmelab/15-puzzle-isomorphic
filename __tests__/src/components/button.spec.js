import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Button from '../../../src/components/button';

describe('Button', () => {
    test('should render the Button without error', () => {
        renderer.create(
            <Button icon="" label="" route="" onClick={() => {}} />,
        );
    });

    test('should render the Button correctly', () => {
        const component = renderer.create(
            <Button icon="" label="" route="" onClick={() => {}} />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should contain a link with', () => {
        const app = shallow(<Button icon="icon" label="label" route="path" />);
        expect(app.find('a').length).toEqual(1);
    });

    test('should render the icon', () => {
        const app = shallow(<Button icon="icon" label="label" route="path" />);
        expect(app.find('i').text()).toEqual('icon');
    });

    test('should contain a link even if there is not route', () => {
        const app = shallow(
            <Button icon="icon" label="label" onClick={() => {}} />,
        );
        expect(app.find('a').length).toEqual(1);
    });

    test('should render the icon even if there is not route', () => {
        const app = shallow(
            <Button icon="icon" label="label" onClick={() => {}} />,
        );
        expect(app.find('i').text()).toEqual('icon');
    });
});
