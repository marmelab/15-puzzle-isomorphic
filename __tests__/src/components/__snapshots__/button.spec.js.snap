// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the Button correctly 1`] = `
<a
  className="btn"
  href=""
  onClick={[Function]}
>
  <i
    className="material-icons left"
  >
    
  </i>
  
</a>
`;
